src/utils/create-facttable.ts

```diff
@@ -0,0 +1,31 @@
+import { Database } from 'duckdb-async';
+import { Dataset } from '../entities/dataset/dataset';
+import { Revision } from '../entities/dataset/revision';
+import { duckdb } from '../services/duckdb';
+import { createEmptyFactTableInCube, loadFactTables } from '../services/cube-handler';
+import { logger } from './logger';
+import { FileValidationErrorType, FileValidationException } from '../exceptions/validation-exception';
+
+export async function createEmptyCubeWithFactTable(dataset: Dataset): Promise<Database> {
+  let endRevision: Revision;
+  if (dataset.draftRevision) {
+    endRevision = dataset.draftRevision;
+  } else if (dataset.publishedRevision) {
+    endRevision = dataset.publishedRevision;
+  } else {
+    throw new Error('No revision present on the dataset');
+  }
```

Comment:
FYI, there's now a `dataset.endRevision` shortcut that can be hydrated that should be set appropriately depending on if there's an active draft or not.

Reply:
Changed this logic to just use `dataset.endRevision` and throw an exception if its not populated